vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet_unittest.cc

```diff
@@ -5,51 +5,85 @@
 
 #include "bat/ledger/internal/wallet/wallet.h"
 
+#include <utility>
 #include <vector>
 
 #include "bat/ledger/internal/core/bat_ledger_test.h"
+#include "bat/ledger/internal/endpoint/promotion/promotions_util.h"
 #include "bat/ledger/internal/ledger_impl.h"
 #include "bat/ledger/internal/state/state_keys.h"
+#include "net/http/http_status_code.h"
 
 namespace ledger {
 
-class WalletTest : public BATLedgerTest {};
+class WalletTest : public BATLedgerTest {
+ protected:
+  mojom::Result CreateWalletIfNecessary() {
+    auto* ledger = GetLedgerImpl();
+
+    auto response = mojom::UrlResponse::New();
+    response->status_code = net::HTTP_CREATED;
+    response->body =
+        "{\"paymentId\": \"37742974-3b80-461a-acfb-937e105e5af4\"}";
+
+    AddNetworkResultForTesting(
+        endpoint::promotion::GetServerUrl("/v3/wallet/brave"),
+        mojom::UrlMethod::POST, std::move(response));
+
+    base::RunLoop run_loop;
+    mojom::Result result;
+    ledger->wallet()->CreateWalletIfNecessary([&result, &run_loop](auto r) {
+      result = r;
+      run_loop.Quit();
+    });
+
+    run_loop.Run();
+    return result;
+  }
+};
 
 TEST_F(WalletTest, GetWallet) {
   auto* ledger = GetLedgerImpl();
+  bool corrupted = false;
 
-  // A wallet is created if none exists
-  ledger->ledger_client()->SetStringState(state::kWalletBrave, "");
-  auto wallet = ledger->wallet()->GetWallet(true);
-  ASSERT_TRUE(wallet);
-  EXPECT_TRUE(wallet->payment_id.empty());
-  EXPECT_TRUE(!wallet->recovery_seed.empty());
-
-  // The created wallet is saved
-  std::vector<uint8_t> recovery_seed = wallet->recovery_seed;
-  wallet = ledger->wallet()->GetWallet(true);
-  ASSERT_TRUE(wallet);
-  EXPECT_EQ(wallet->recovery_seed, recovery_seed);
+  // When there is no current wallet information, `GetWallet` returns empty and
+  // sets the corrupted flag to false.
+  GetTestLedgerClient()->SetStringState(state::kWalletBrave, "");
+  corrupted = true;
```

Comment:
Total nit, but seems like you could just initialize `corrupted` to `true` up above (unless you're trying to draw attention to this assignment here, not sure).